packages/payments-plugin/e2e/paypal-dev-server.ts

```diff
@@ -0,0 +1,118 @@
+    if (!PAYPAL_CLIENT_ID || !PAYPAL_CLIENT_SECRET || !PAYPAL_MERCHANT_ID) {
+        throw new Error('Please provide PayPal credentials in the .env file');
+    }
+
+    const apiUrl = 'https://api-m.sandbox.paypal.com';
+
+    registerInitializer('sqljs', new SqljsInitializer(path.join(__dirname, '__data__')));
+    const config = mergeConfig(testConfig, {
+        plugins: [
+            ...testConfig.plugins,
+            DefaultSearchPlugin,
+            AdminUiPlugin.init({
+                route: 'admin',
+                port: 5001,
+            }),
+            PayPalPlugin.init({
+                apiUrl,
+            }),
+            PayPalSdkPlugin.init({
+                apiUrl,
+                clientId: PAYPAL_CLIENT_ID,
+            }),
+        ],
+        logger: new DefaultLogger({ level: LogLevel.Debug }),
+        apiOptions: {
+            adminApiPlayground: true,
+            shopApiPlayground: true,
+        },
+    });
+    const { server, adminClient, shopClient } = createTestEnvironment(config as any);
+    await server.init({
+        initialData,
+        productsCsvPath: path.join(__dirname, 'fixtures/e2e-products-minimal.csv'),
+        customerCount: 1,
+    });
+    // Set EUR as currency for PayPal
+    await adminClient.asSuperAdmin();
+    await adminClient.query(gql`
+        mutation {
+            updateChannel(input: { id: "T_1", currencyCode: EUR }) {
+                __typename
+            }
+        }
+    `);
+    // Create method
+    await adminClient.query<CreatePaymentMethodMutation, CreatePaymentMethodMutationVariables>(
+        CREATE_PAYMENT_METHOD,
+        {
+            input: {
+                code: 'paypal',
+                translations: [
+                    {
+                        languageCode: LanguageCode.en,
+                        name: 'PayPal payment test',
+                        description: 'This is a PayPal test payment method',
+                    },
+                ],
+                enabled: true,
+                handler: {
+                    code: 'paypal',
+                    arguments: [
+                        { name: 'clientId', value: PAYPAL_CLIENT_ID },
+                        { name: 'clientSecret', value: PAYPAL_CLIENT_SECRET },
+                        { name: 'merchantId', value: PAYPAL_MERCHANT_ID },
+                    ],
+                },
+            },
+        },
+    );
+    // Prepare a test order where the total is 0
+    await shopClient.asUserWithCredentials('hayden.zieme12@hotmail.com', 'test');
+    await shopClient.query(ADD_ITEM_TO_ORDER, {
+        productVariantId: 'T_1',
+        quantity: 1,
+    });
+    await setShipping(shopClient);
+    await shopClient.query(APPLY_COUPON_CODE, { couponCode: 'FREE_SHIPPING' });
+
+    await shopClient.query(TRANSITION_TO_STATE, {
+        state: 'ArrangingPayment',
+    });
+
+    // eslint-disable-next-line no-console
+    console.log('Auth Token: ', shopClient.getAuthToken());
```

Comment:
This token is printed, so we can enter it in the Web UI of the paypal plugin. I would be open to setting this automatically in the Vendure REST Endpoint, but I wasn't quite sure how.